src/components/shared/subscribe/subscribe.jsx

```diff
@@ -26,22 +33,51 @@ const links = [
   },
 ];
 
+const appearAndExitAnimationVariants = {
+  initial: { opacity: 0 },
+  animate: { opacity: 1, transition: { duration: 0.2 } },
+  exit: { opacity: 0, transition: { duration: 0.2 } },
+};
+
 const Subscribe = () => {
   const [email, setEmail] = useState('');
+  const [formState, setFormState] = useState('default');
+  const [errorMessage, setErrorMessage] = useState('');
 
   const handleInputChange = (event) => setEmail(event.currentTarget.value.trim());
 
   const handleSubmit = (event) => {
     event.preventDefault();
 
-    fetch('https://submit-form.com/nHIBlORO', {
-      headers: {
-        'Content-Type': 'application/json',
-        Accept: 'application/json',
-      },
-      method: 'POST',
-      body: JSON.stringify({ email }),
-    });
+    if (!email) {
+      setErrorMessage('Please enter your email');
+    } else if (!emailRegexp.test(email)) {
+      setErrorMessage('Please valid email');
+    } else {
+      setErrorMessage('');
+      setFormState('loading');
+
+      setTimeout(() => {
+        setFormState('success');
+        setEmail('Thanks for subscribing!');
+
+        setTimeout(() => {
+          setFormState('default');
+          setEmail('');
+        }, 2000);
+
+        // 2000 (loading animation duration) + 200 (loading animation delay) = 2200
+      }, 2200);
+
+      fetch('https://submit-form.com/nHIBlORO', {
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+        },
+        method: 'POST',
+        body: JSON.stringify({ email }),
+      });
```

Comment:
So we are setting `success` state _before_ we actually submitted form data? What if it would fail? 🤔  

I think we have to wrap `fetch` in try/catch or use promise chain and set those timeouts _after_ the post request has succeeded, or display eror.